src/App.tsx

```diff
@@ -1,23 +1,310 @@
+        return (<div className="header">
+            <Link to="/"><img className="header__logo" src={logo} alt="" /></Link>
+            <button onClick={(e) => { setInputActive(!inputActive) }} className="header__search-button header__search-button_visible"></button>
+            <input onKeyDown={(e) => e.key === "Enter" && navigate('/search', { state: e.currentTarget.value })} id="header__search-field" className={inputActive ? "header__search-field_visible" : "header__search-field"} type="text" />
+        </div>);
+    else
+        return (<div className="header">
+            <Link to="/"><img className="header__logo" src={logo} alt="" /></Link>
+        </div>);
+}
+function Heading(props: { text: string }) {
+    return (<div className="heading">
+        <h2 className="heading__text text">{props.text}</h2>
+        <hr className="heading__separator" />
+    </div>)
+}
+function Peformers() {
+    const [perfromersList, setPerfromersList] = useState<Data[]>([]);
+    useEffect(() => {
+        getTopArtists(api_key)
+            .then(perfromersList => setPerfromersList(perfromersList))
+    }, []);
+    if (perfromersList.length === 0) { return null; }
+    return (<div className="performers">
+        <HeadlinePerformers data={perfromersList} />
+        <RegularPerformers data={perfromersList} />
+    </div>)
+}
+function HeadlinePerformer(props: { data: Data }) {
+    const image = props.data.image[4]["#text"]
+    const divStyle = {
+        background: `linear-gradient(to top, black -50%, transparent 50%), url(${image})`,
+    };
+    return (<a className="headline-performers__performer" href={props.data.url}>
+        <div className="headline-performers__image" style={divStyle} >
+            <h3 className="headline-performers__name text">
+                {props.data.name}
+            </h3>
+        </div>
+    </a>)
+}
+function HeadlinePerformers(props: { data: Array<Data> }) {
+    return (<div className="headline-performers">
+        <HeadlinePerformer data={props.data[0]} />
+        <HeadlinePerformer data={props.data[1]} />
+    </div>)
+}
+function RegularPerformer(props: { data: Data }) {
+    const image = props.data.image[4]["#text"]
+    return (<a className="regular-performers__performer" href={props.data.url}>
+        <img className="regular-performers__image" src={image} alt="" />
+        <h3 className="regular-performers__name text">{props.data.name}</h3>
+    </a>);
+}
+function RegularPerformers(props: { data: Array<Data> }) {
+    const performers = props.data.slice(2, 14)
+    return (<div className="regular-performers">
+        {performers.map((item) => { return <RegularPerformer data={item} key={item.name} /> })}
+    </div>);
+}
+function TrackList() {
+    const [trackList, setTrackList] = useState<Data[]>([]);
+    useEffect(() => {
+        getTopTracks(api_key)
+            .then(trackList => setTrackList(trackList))
+    }, []);
+    if (trackList.length === 0) { return null; }
```

Comment:
есть вот такая клевая штука, это конечно, так называемый "синтаксический сахар". НО сильно упрощает написание кода 

```suggestion
    if (trackList?.length === 0) { return null; }
```
https://learn.javascript.ru/optional-chaining